src/query-runner/BaseQueryRunner.ts

```diff
@@ -137,6 +147,28 @@ export abstract class BaseQueryRunner {
         // Do nothing
     }
 
+    /**
+     * add after rollback callback. If transaction is not active, executes callback immediately.
+     */
+    async onAfterCommit(callback: Function): Promise<void> {
+        if(this.isTransactionActive) {
+            this.afterCommitListeners.push(callback);
+        } else {
+            await callback();
+        }
+    }
```

Comment:
Do we really want to execute callback immediately when transaction is not active or throw an error like `rollbackTransaction()` would?
I personally would expect this to be executed only after a commit so not when there is none. Throwing error would prevent calling `onAfterCommit` too early, before startTransaction

Reply:
@jpagand The goal of functions I've added is to run the code without knowing if you are in a transaction or not.

I can rename it to `onAfterCommitOrNow` :D

Reply:
If I want to execute a callback on after commit I really don't want it to be executed now otherwise I would just call it.
Can you give a use case where you would want a such behavior?
As a dev, `onAfterCommitOrNow` sounds very flaky as I don't know exactly when it would be executed

Reply:
Into the bookableUnit resolver (create or update) i don't know if I am into a transaction or not but I want the postMessageToAutomationQueue to be executed after the commit or now

Reply:
I'm a bit lost, what is this? It's a workaround of what?